src/lib/wallet.ts

```diff
@@ -443,24 +444,22 @@ export const usePublishedKeys = (path: string) => {
 export const usePublishedDatum = (path: string) => {
   const [status, setStatus] = useState(LoadStatus.Idle);
   const [data, setData] = useState({} as any);
-  const walletUtils = useContext(WalletContext);
 
   // XXX cleanup? await next?
   useEffect(() => {
-    const { follow } = rpcUtils;
-    const fetchData = async () => {
-      console.debug('usePublishedDatum following', `:published.${path}`);
-      const follower = await follow(`:published.${path}`);
-      const iterable: AsyncIterable<Record<string, unknown>> =
-        await follower.getLatestIterable();
-      const iterator = iterable[Symbol.asyncIterator]();
-      setStatus(LoadStatus.Waiting);
-      const { value: publishedValue } = await iterator.next();
-      setData(publishedValue.value);
-      setStatus(LoadStatus.Received);
-    };
-    fetchData().catch(e => console.error('useEffect error', e));
-  }, [path, walletUtils]);
+    const { storageWatcher } = rpcUtils;
+
+    setStatus(LoadStatus.Waiting);
+
+    return storageWatcher.watchLatest(
```

Comment:
much more readable.

imo the `XXX cleanup?` comment is done

Reply:
Oh true, nice